src/lib/components/EventNetwork.svelte

```diff
@@ -0,0 +1,395 @@
+<script lang="ts">
+  import { onMount } from "svelte";
+  import * as d3 from "d3";
+  import type { NDKEvent } from "@nostr-dev-kit/ndk";
+
+  export let events: NDKEvent[] = [];
+
+  let svg;
+  let isDarkMode = false;
+  const width = 1200;
+  const height = 600;
+  const nodeRadius = 20;
+  const dragRadius = 45;
+  const linkDistance = 120;
+
+  function updateTheme() {
+    isDarkMode = !isDarkMode;
+    document.body.classList.toggle("dark");
+    if (svg) drawNetwork();
+  }
+
+  function getEventColor(eventId: string): string {
+    const num = parseInt(eventId.slice(0, 4), 16);
+    const hue = num % 360;
+    const saturation = 70;
+    const lightness = 75;
+    return `hsl(${hue}, ${saturation}%, ${lightness}%)`;
+  }
+
+  function generateGraph(events: NDKEvent[]) {
```

Comment:
This function returns a tuple.  Let's annotate that as so:

```suggestion
  function generateGraph(events: NDKEvent[]): [object[], object[]] {
```

Ideally, we define types or interfaces for the objects contained in `nodes` and `links`, and use those in place of `object` in the return type annotation.